tsl/test/expected/dist_hypertable-11.out

```diff
@@ -4763,3 +4763,554 @@ SET timescaledb.enable_per_data_node_queries = true;
 SELECT tableoid::regclass, * FROM dist_syscol;
 ERROR:  system columns are not accessible on distributed hypertables with current settings
 \set ON_ERROR_STOP 1
+-- Test SkipScan
+-- Add an index on the value column
+CREATE TABLE hyper (time bigint NOT NULL, device integer);
+CREATE INDEX idx_hyper_dev ON hyper (device);
+SELECT create_distributed_hypertable('hyper', 'time',  chunk_time_interval => 200);
```

Comment:
I a wondering how push-down works for a space-partitioned table, which is actually the recommended way to partition. For example:

```
SELECT create_distributed_hypertable('hyper', 'time', 'device', chunk_time_interval => 200);
```

Ideally, if we do `SELECT DISTINCT device` on such a table, we shouldn't need to run a HashAgg or Unique node on the access node since we know that the responses from the data nodes are unique already (given that we spread the data across data nodes according to the "device" key"). So, it would be sufficient to just append the results from each data node, which would be much more efficient.

So, the intuition would be that if the DISTINCT covers the _all_ partitioning keys OR the _space_ dimension (if such a dimension exists), then the query requires no unique processing on the data node.

Similarly, if you partition only on time, like in the test here, you should be able to push down `SELECT DISTINCT time` without unique processing on the access node. But I don't see such a test.

Looking at the code, I don't see any checks that compare the distinct keys to the partitioning keys, so I suspect eliding unique processing on the access node is not implemented. I guess that's OK for the initial version (should still give a good performance boost). But, ideally, I think we should try to avoid, e.g., doing a HashAgg on the access node when we don't need it to make the best use of multi-node. Obviously, this further improvement could be done separately, but I am mostly wondering if this was a conscious decision or the partition key coverage case wasn't considered?


I would at least like to see a `SELECT DISTINCT time` query here.

Reply:
Checks that compare distinct keys with space-time partitioned columns will be added as a later optimization. I wanted to focus on the pushdown of SELECT DISTINCT in this patch and get that right first.

Reply:
Fair enough. Please make sure there's an issue for this further refinement though.

Reply:
#3121 opened to track this refinement